web/packages/teleterm/src/ui/services/resources/resourcesService.ts

```diff
@@ -117,6 +125,25 @@ export class AmbiguousHostnameError extends Error {
   }
 }
 
+export class ResourceSearchError extends Error {
+  clusterUri: uri.ClusterUri;
+  resourceKind: SearchResult['kind'];
+
+  constructor(
+    clusterUri: uri.ClusterUri,
+    resourceKind: SearchResult['kind'],
+    cause: Error
+  ) {
+    super(
+      `Error while fetching resources of type ${resourceKind} from a cluster`,
+      { cause }
```

Comment:
This the cause https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#the-usedefineforclassfields-flag-and-the-declare-property-modifier 

The solution is to either add `declare` to the `cause` field (but our Babel config doesn't support it 🙄) or just recreate the field:
```ts
export class ResourceSearchError extends Error {
  constructor(
    private clusterUri: uri.ClusterUri,
    private resourceKind: SearchResult['kind'],
    public cause: Error
  ) {
    super(
      `Error while fetching resources of type ${resourceKind} from a cluster`,
    );
    this.name = 'ResourceSearchError';
  }
```

Here I used a shorter syntax (`private/public` in the constructor) that creates a class field with an assigned value.

Reply:
I feel like I'd rather just keep it as it is and keep `cause` as a regular `Error` field.

Reply:
IMO types safety outweighs the disadvantage (a little tricky way with redeclaring the field), but I don't insist on that.

Anyway, we can still use `private` keyword for `clusterUri` and `resourceUri` so we don't have to assign the values in `constructor`, WDYT?

Reply:
> Anyway, we can still use `private` keyword for `clusterUri` and `resourceUri` so we don't have to assign the values in `constructor`, WDYT?

I'd rather wait with that since reading those fields might be useful for error handling.